http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -127,6 +127,52 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
     } yield (callerActions, projectCallerActions)
   }
 
+  /**
+   * Get all Sam resource IDs of the given type on which the user is granted permission to `discover`. Returns ids with explicitly
+   * granted roles only; users with workspace `owner` role, for example, can see all runtimes in the workspace, but
+   * `listResourceIds[SamRuntimeResourceId]` is not guaranteed to return all runtimes in that workspace. Works for any
+   * Sam resource type except SharedApp and similar alternative app types; these require a specific Sam resource object
+   * to determine permissions, as these are stored on the individual records.
+   * @param hasOwnerRole whether to return only resources that the user is granted ownership on (has role = `ownerRoleName`)
+   * @param userInfo the current user
+   * @param resourceDefinition the resource archetype (App, Runtime, etc)
+   * @param resourceIdDecoder converts Json -> SamResourceId, needed by SamDao
+   * @param ev the request context
+   * @tparam R SamResourceId
+   * @return List[SamResourceId] of all resources of type R which are discoverable (or owned, if `hasOwnerRole`) by the user
+   */
+  override def listResourceIds[R <: SamResourceId](
+    hasOwnerRole: Boolean,
+    userInfo: UserInfo
+  )(implicit
+    resourceDefinition: SamResource[R],
+    appDefinition: SamResource[AppSamResourceId],
+    resourceIdDecoder: Decoder[R],
+    ev: Ask[F, TraceId]
+  ): F[Set[R]] = {
+    require(resourceDefinition != appDefinition)
+    val authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
+
+    // TODO get `ownerRoleName` from Sam config for the resource type
```

Comment:
What does this TODO mean? That's what the code is doing, right -- looking up the ownerRole from our modeling of the different Sam resource types in https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoAuthProvider.scala#L37

Reply:
Right, it's looking it up in LeoAuthProvider, not Sam. Sam should be the point of truth for all this - I think that's in the Config API under `/resourceTypes`; all the types are listed there with `ownerRoleName` etc.

Reply:
I mean, I think Leo will still need to have knowledge of the exact Sam resource types (and roles, actions) it's using, and how those map to user actions in Leo. Not sure it needs to dynamically pull that from the `Config` API. I think what we should strive for is to a better of job of encoding Leo's access model in Sam, so we don't have to merge all these permissions from separate checks on project/notebook-cluster/workspace/controlled-resource resources.

Reply:
I see that Leo needs to know which Sam resource types and actions it uses (and roles, tho I hope we could model most stuff as action + resource type). I agree that once the Sam model reflects the actual Leo access we want, we shouldn't need to look at the configuration at all, because checking action+type with Sam would be enough.

I can rm the todo if you suggest! I didn't want to leave `ownerRoleName` etc uncommented, since it's definitely not what we want to do in any case. Leo should not need to know.

Reply:
I changed the TODO!